### PR TITLE
Feat: add Y,YMD,YM mode to calender

### PR DESCRIPTION
**Pull request by shayan-hamidi**

### CODE REVIEW

File: packages/core/organisms/PersianCalendar/PersianCalendar.vue [hunk 1]
```diff
@@ -4,17 +4,34 @@ import NextMonthArrow from './components/NextMonthArrow.vue'
...
+    type: String as PropType<'default' | 'YMD' | 'YM' | 'Y'>,
+    default: () => 'YMD'
+  },
+  yearPeriod: {
+    type: Array as PropType<number[]>,
+    default: () => [1320, 1402]
   }
 })
-
+const currentYear = ref<number>(1402)
```
Comment (ialiaslani) [inline packages/core/organisms/PersianCalendar/PersianCalendar.vue]:
```suggestion
const currentYear = ref<number>(gorgianToJalali(new Date())[0])
```

File: packages/core/organisms/PersianCalendar/PersianCalendar.vue [hunk 2]
```diff
@@ -90,6 +167,160 @@ const selectingDate = (day: number) => {
...
+            <PersianTypography
+              :text="`${yearPeriodRange[yearPeriodRange.length - 1]} - ${yearPeriodRange[0]}`"
+            />
+          </div>
+          <div v-else-if="steps.month" class="cursor-pointer">
+            <PersianTypography :text="String(currentYear)" />
+          </div>
+          <div v-if="steps.day" class="flex justify-center">
+            <a class="mx-1 cursor-pointer" :style="{ color: 'black' }" :onclick="monthStepShortCut">
+              <PersianTypography :text="monthName.split(' ')[0]" />
```
Comment (ialiaslani) [inline packages/core/organisms/PersianCalendar/PersianCalendar.vue:221]:
```suggestion
              <PersianTypography :text="currentMonthName" />
```
and a computed like `const curerntMonthName = computed(() => monthName.value.split(' ')[0])`

File: packages/core/organisms/PersianCalendar/PersianCalendar.vue [hunk 3]
```diff
@@ -90,6 +167,160 @@ const selectingDate = (day: number) => {
...
+          <div v-if="steps.year">
+            <PersianTypography
+              :text="`${yearPeriodRange[yearPeriodRange.length - 1]} - ${yearPeriodRange[0]}`"
+            />
+          </div>
+          <div v-else-if="steps.month" class="cursor-pointer">
+            <PersianTypography :text="String(currentYear)" />
+          </div>
+          <div v-if="steps.day" class="flex justify-center">
+            <a class="mx-1 cursor-pointer" :style="{ color: 'black' }" :onclick="monthStepShortCut">
```
Comment (ialiaslani) [inline packages/core/organisms/PersianCalendar/PersianCalendar.vue]:
```suggestion
            <PersianButton class="mx-1 cursor-pointer" :style="{ color: 'black' }" @click="monthStepShortCut">
```

File: packages/core/organisms/PersianCalendar/PersianCalendar.vue [hunk 4]
```diff
@@ -90,6 +167,160 @@ const selectingDate = (day: number) => {
...
+              <PersianTypography :text="monthName.split(' ')[0]" />
+            </a>
+            <a class="mx-1 cursor-pointer" :style="{ color: 'black' }" :onclick="toYearStep">
+              <PersianTypography :text="monthName.slice(-4)" />
```
Comment (ialiaslani) [inline packages/core/organisms/PersianCalendar/PersianCalendar.vue:224]: use computed as well

File: packages/core/organisms/PersianCalendar/PersianCalendar.vue [hunk 5]
```diff
@@ -90,6 +167,160 @@ const selectingDate = (day: number) => {
...
+        <div
+          v-if="steps.year"
+          class="grid grid-cols-3 gap-x-[30px] gap-y-[10px]"
+          :style="{ direction: 'ltr' }"
+        >
+          <PersianButton
+            v-for="year in yearsList"
+            :key="year"
+            variant="transparent"
+            :onClick="() => toMonthStep(year)"
```
Comment (ialiaslani) [inline packages/core/organisms/PersianCalendar/PersianCalendar.vue]:
```suggestion
            @click="() => toMonthStep(year)"
```

File: packages/core/organisms/PersianCalendar/PersianCalendar.vue (hunk 2 above)
Comment (shayan-hamidi) [reply packages/core/organisms/PersianCalendar/PersianCalendar.vue:221]: Property 'split' does not exist on type 'ComputedRef<string>'

**Pull request merged**